From 64a76f3ef625aa8b8cb614cb0220e9c026e10651 Mon Sep 17 00:00:00 2001
From: Juan Antonio Gozalvez Herrero <jgozalvx>
Date: Tue, 18 Sep 2012 10:30:41 +0200
Subject: [PORT FROM MAIN] Revert "WLAN: Skip bind/unbind when no Wi-Fi module present"

This reverts commit ffa45a438bf72f2d4c7d3fcfacbf21344a02dcb0.
This reverts patch : http://android.intel.com:8080/#/c/64718/

Change-Id: I249d95a5b741c38ce1132b3c9a0f716483ea350f
Orig-Change-Id: I35feaa939f8c8e3108e8c9638956682a87b99043
Signed-off-by: Juan Antonio Gozalvez Herrero <jgozalvx>
Reviewed-on: http://android.intel.com:8080/66868
Reviewed-by: Gozalvez Herrero, Juan AntonioX <juan.antoniox.gozalvez.herrero@intel.com>
Tested-by: Gozalvez Herrero, Juan AntonioX <juan.antoniox.gozalvez.herrero@intel.com>
Reviewed-on: http://android.intel.com:8080/77983
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
---
 wlan_prov/wlan_provisioning.c |    8 ++------
 1 files changed, 2 insertions(+), 6 deletions(-)

diff --git a/wlan_prov/wlan_provisioning.c b/wlan_prov/wlan_provisioning.c
index 9531497..a79513b 100644
--- a/wlan_prov/wlan_provisioning.c
+++ b/wlan_prov/wlan_provisioning.c
@@ -245,11 +245,6 @@ int main(int argc, char **argv)
 		return -1;
 	}
 
-       if (sdio_get_pci_id(SYSFS_SDIO_DEVICES_PATH, device_id)) {
-		LOGE("no wlan device detected, exit...");
-		return -1;
-       }
-
 #ifdef BUILD_WITH_CHAABI_SUPPORT
 	/* Read MAC address from Chaabi */
 	if (get_customer_data(ACD_WLAN_MAC_ADDR_FIELD_INDEX,
@@ -328,7 +323,8 @@ end:
 		* account. In that case, the reboot is required after flashing for the
 		* time the board.
 		*/
-		if (unbind_wlan_sdio_drv(WLAN_SDIO_BUS_PATH, device_id)
+		if (sdio_get_pci_id(SYSFS_SDIO_DEVICES_PATH, device_id)
+			|| unbind_wlan_sdio_drv(WLAN_SDIO_BUS_PATH, device_id)
 			|| bind_wlan_sdio_drv(WLAN_SDIO_BUS_PATH, device_id)) {
 			/*
 			* Rebooting the board: In this level, the NVS was saved. The
-- 
1.7.4.1

